sdk/web-pubsub/web-pubsub/src/utils.ts

```diff
@@ -0,0 +1,28 @@
+import { HttpRequestBody } from "@azure/core-http";
+
+function isHttpRequestBody(obj: unknown): obj is HttpRequestBody {
+  return (
+    typeof obj === "function" ||
+    (typeof obj === "object" &&
+      obj != null &&
+      (obj.constructor.name === "ArrayBuffer" ||
+        obj.constructor.name === "Blob" ||
+        ArrayBuffer.isView(obj)))
+  );
+}
+
+export function getContentTypeForMessage(
+  message: unknown,
+  options: Record<string, any>
+): "text/plain" | "application/json" | "application/octet-stream" {
+  if (options?.contentType === "text/plain") {
+    if (typeof message !== "string") {
+      throw new TypeError("Message must be a string.");
```

Comment:
this feels weird... we're type-checking inside a method called `getContentType` and using options passed in to decide the content type instead of the message itself?

Reply:
The message itself isn't enough - a string can either be a text/plain or a JSON string.